evalml/data_checks/datetime_nan_data_check.py

```diff
@@ -0,0 +1,57 @@
+from evalml.data_checks import DataCheck, DataCheckError, DataCheckMessageCode
+from evalml.utils.woodwork_utils import (
+    _convert_woodwork_types_wrapper,
+    infer_feature_types
+)
+
+error_contains_nan = "Input datetime column ({}) contains NaN values. Please input NaN values or drop this column."
+
+
+class DateTimeNaNDataCheck(DataCheck):
```

Comment:
Do we have a standardization of NaN vs Null terminology? I decided to use NaN here but can change to null as well.

Reply:
That's a good question...I don't *think* nulls fall into the scope of this.  We have data checks to alert us to highly null columns and I don't think a null by itself makes the algorithm choke.